cocos/2d/CCSprite.cpp

```diff
@@ -417,8 +417,11 @@ void Sprite::setProgramState(backend::ProgramState *programState)
 
 void Sprite::setTexture(Texture2D *texture)
 {
-    auto isETC1 = texture && texture->getAlphaTextureName();
-    setProgramState((isETC1) ? backend::ProgramType::ETC1 : backend::ProgramType::POSITION_TEXTURE_COLOR);
+    if (_programState == nullptr || _programState->getProgram()->getProgramType() == backend::ProgramType::POSITION_TEXTURE_COLOR) {
+        const auto isETC1 = texture && texture->getAlphaTextureName();
+        setProgramState((isETC1) ? backend::ProgramType::ETC1 : backend::ProgramType::POSITION_TEXTURE_COLOR);
+    }
```

Comment:
Thank you for your advice.

Is it okay with such a fix?
```suggestion
    if (_programState == nullptr || _programState->getProgram()->getProgramType() == backend::ProgramType::POSITION_TEXTURE_COLOR ||
        _programState->getProgram()->getProgramType() == backend::ProgramType::ETC1) {
        const auto isETC1 = texture && texture->getAlphaTextureName();
        setProgramState((isETC1) ? backend::ProgramType::ETC1 : backend::ProgramType::POSITION_TEXTURE_COLOR);
    }
```

Reply:
Yes that would work fine, since it covers all cases.